broadinstitute/rawls
# GAWB-603: Use typesafe config correctly so we can easily override settings

This change renames application.conf to reference.conf and loads rawls.conf using a java option parameter so we don't have to hard code the path. This gives us the flexibility to change configs very easily by just overriding the value in rawls.conf and rebooting rawls.

Tested this out as follows:

1) Edit spray timeout to be 1 sec in reference.conf
2) Verify that listWorkspaces times out
3) Override spray timeout in rawls.conf to be 10 sec
4) Reboot rawls
5) Verify that listWorkspaces doesn't time out after 1 second

Also see:

https://github.com/broadinstitute/firecloud-develop/pull/182
https://github.com/broadinstitute/firecloud-develop/pull/183

  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
